File: CaneOrbitCsharp.Api/Models/Field.cs
namespace CaneOrbis.Api.Models
{
    public class Field
    {
        public int IdField { get; set; }

        public int IdPropriedade { get; set; }

        public int? IdEosField { get; set; }

        public string NmField { get; set; } = string.Empty;

        public decimal? VlAreaHectare { get; set; }

        public DateTime DtCriacao { get; set; } = DateTime.Now;

        public Propriedade? Propriedade { get; set; }

        public ICollection<DispositivoIot> DispositivosIot { get; set; } = new List<DispositivoIot>();
    }
}

File: caneorbitJava/src/main/java/br/com/fiap/tdspo/gsolution/caneorbit/domain/model/Field.java
package br.com.fiap.tdspo.gsolution.caneorbit.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "T_ORB_FIELD")
public class Field {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_FIELD", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ID_PROPRIEDADE", nullable = false)
    private Propriedade propriedade;

    @Column(name = "ID_EOS_FIELD")
    private Long idEosField;

    @Column(name = "NM_FIELD", nullable = false, length = 100)
    private String nome;

    @Column(name = "VL_AREA_HECTARE", precision = 10, scale = 2)
    private BigDecimal areaHectare;

    @Column(name = "DT_CRIACAO", nullable = false)
    private LocalDateTime dataCriacao;

    @OneToMany(mappedBy = "field", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<DispositivoIot> dispositivos = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        this.dataCriacao = LocalDateTime.now();
    }
}